### PR TITLE
fix(route): finviz news

### LINKED ISSUE #12836: finviz时间
### 路由地址

```routes
/finviz/:category?
```

### 完整路由地址

```fullroutes
/finviz/:category?
```

### 相关文档

https://docs.rsshub.app/en/finance.html#finviz-news

### 预期是什么？

时间获取正常。

### 实际发生了什么？

解析的时间多24h。

https://github.com/DIYgod/RSSHub/blob/9e06d3ab93b5ee4d552d133e442acf9939ec79a3/lib/v2/finviz/news.js#L31

可能未考虑天数不同。

### 部署

RSSHub 演示 (https://rsshub.app)

### 部署相关信息

_No response_

### 额外信息

```shell
.
```

### 这不是重复的 issue

- [X] 我已经搜索了[现有 issue](https://github.com/DIYgod/RSSHub/issues)，以确保该错误尚未被报告。

### PR DESCRIPTION
**Pull request by nczitzk**
## 该 PR 相关 Issue / Involved Issue

Close #12836

## 路由地址示例 / Example for the Proposed Route(s)

```routes
/finviz/news
/finviz/blogs
```

## 新 RSS 路由检查表 / New RSS Route Checklist
  
- [ ] 新的路由 New Route
  - [ ] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [ ] 文档说明 Documentation
  - [ ] 中文文档 CN
  - [ ] 英文文档 EN
- [ ] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/v2/finviz/news.js [hunk 1]
```diff
@@ -3,43 +3,68 @@ const cheerio = require('cheerio');
...
-                pubDate: timezone(parseDate(item.find('.nn-date').text(), ['MMM-DD', 'HH:mmA']), -4),
-                description:
-                    item
-                        .find('td[data-boxover]')
-                        .attr('data-boxover')
-                        ?.match(/<td class='news_tooltip-tab'>([\s\S]*)<\/td>/)[1] ?? '',
-            };
+            const a = item.find('a.nn-tab-link');
+
+            return ctx.cache.tryGet(a.prop('href'), () => {
```
Comment (TonyRL) [inline lib/v2/finviz/news.js:36]: Some of the `(a.prop('href')` are `undefined`

Comment (nczitzk) [reply lib/v2/finviz/news.js:36]: Unable to reproduce locally, I think I'll remove support for caching.

Comment (TonyRL) [reply lib/v2/finviz/news.js]: I guess it's caused by Ads which are only visible to RSSHub.


<details><summary>Feed screenshot</summary>

image
</details>

Which can be solved by:
```diff
            const descriptionMatches = a
                .parent()
                .prop('data-boxover')
-                .match(/<td class='news_tooltip-tab'>(.*?)<\/td>/);
+                ?.match(/<td class='news_tooltip-tab'>(.*?)<\/td>/);
            const authorMatches = item
                .find('use')
                .first()
                .prop('href')
-                .match(/#(.*?)-(light|dark)/);
+                ?.match(/#(.*?)-(light|dark)/);
```
and a filter like `.filter((item) => item.title)`

Comment (nczitzk) [reply lib/v2/finviz/news.js]: thanks. 😄

**Pull request merged**